cpp/src/Slice/SliceUtil.cpp

```diff
@@ -499,30 +510,10 @@ Slice::checkIdentifier(const string& id)
         unit->error("illegal double underscore in identifier `" + name + "'");
         isValid = false;
     }
-    else if(index != string::npos && unit->currentIncludeLevel() == 0 && !unit->allowUnderscore())
+    else if(index != string::npos && unit->currentIncludeLevel() == 0)
```

Comment:
The `&& unit->currentIncludeLevel() == 0` should not be necessary.

Reply:
Also fixed!